File: workspace-import/gradle-plugin/src/com/jetbrains/ls/imports/gradle/model/impl/ModuleSourceSetImpl.java
// Copyright 2000-2026 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
package com.jetbrains.ls.imports.gradle.model.impl;

import com.jetbrains.ls.imports.gradle.model.ModuleSourceSet;
import org.jspecify.annotations.NonNull;

import java.io.File;
import java.util.Set;

public final class ModuleSourceSetImpl implements ModuleSourceSet {

    private final @NonNull String name;
    private final @NonNull Set<@NonNull File> sources;
    private final @NonNull Set<@NonNull File> resources;
    private final @NonNull Set<@NonNull String> excludes;

    public ModuleSourceSetImpl(
            @NonNull String name,
            @NonNull Set<@NonNull File> sources,
            @NonNull Set<@NonNull File> resources,
            @NonNull Set<@NonNull String> excludes
    ) {
        this.name = name;
        this.sources = sources;
        this.resources = resources;
        this.excludes = excludes;
    }

    @Override
    public @NonNull String getName() {
        return name;
    }

    @Override
    public @NonNull Set<File> getSources() {
        return sources;
    }

    @Override
    public @NonNull Set<File> getResources() {
        return resources;
    }

    @Override
    public @NonNull Set<String> getExcludes() {
        return excludes;
    }
}

File: workspace-import/gradle-plugin/src/com/jetbrains/ls/imports/gradle/model/impl/ModuleSourceSetImpl.kt
// Copyright 2000-2026 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
@file:Suppress("IO_FILE_USAGE")

package com.jetbrains.ls.imports.gradle.model.impl

import com.jetbrains.ls.imports.gradle.model.KotlinModule
import com.jetbrains.ls.imports.gradle.model.ModuleJavaSettings
import com.jetbrains.ls.imports.gradle.model.ModuleSourceSet
import java.io.File

data class ModuleSourceSetImpl(
    override val name: String,
    override val sources: MutableSet<File>,
    override val resources: MutableSet<File>,
    override val excludes: Set<String>,
    override val runtimeClasspath: Set<File>,
    override val compileClasspath: Set<File>,
    override val outputDirs: Set<File>,
    override val producedArchives: Set<File>,
    override val friendSourceSets: Set<String>,
    override val hasUnresolvedDependencies: Boolean,
    override val javaSettings: ModuleJavaSettings,
    override val kotlinModule: KotlinModule?
) : ModuleSourceSet
